pkg/controllers/management/eksupstreamrefresh/eks_upstream_refresh.go

```diff
@@ -182,10 +207,26 @@ func GetComparableUpstreamSpec(secretsCache wranglerv1.SecretCache, cluster *mgm
 		}
 
 		nodeGroupStates = append(nodeGroupStates, ng)
+		var nodeGroupMustBeDeleted string
+		if len(ng.Nodegroup.Health.Issues) != 0 {
+			var issueMessages []string
+			for _, issue := range ng.Nodegroup.Health.Issues {
+				issueMessages = append(issueMessages, aws.StringValue(issue.Message))
+				if !controller.NodeGroupIssueIsUpdatable(aws.StringValue(issue.Code)) {
```

Comment:
why not just check if the status is "degraded"?

Reply:
A node group can be degraded for a lot of reasons, and some of them can be taken care of by an update. Here, we are checking to see if the degraded status is unrecoverable.